Swift/Tests/QueryTest.swift

```diff
@@ -1917,26 +1917,28 @@ class QueryTest: CBLTestCase {
         
         let x1 = expectation(description: "wait-for-live-query-changes")
         var count = 0;
+        var docCount = 0;
         let token = query.addChangeListener { (change) in
-            count = count + 1
             let rows =  Array(change.results!)
-            if count == 1 {
-                // initial notification about already existing result set!
-                XCTAssertEqual(rows.count, 9)
-                XCTAssertEqual(rows[0].int(at: 0), 1)
-            } else {
-                // deleted the doc should return empty result set!
-                XCTAssertEqual(rows.count, 8)
+            count = count + rows.count
```

Comment:
I do not understand why the previous code doesn't work. I expected that there will be only 2 callbacks (initial ones and the one after purge).

Reply:
But the callbacks can come in multiple sets right? one with all docIds or two with all docIds. There is no restriction in how the set will be delivered?

I have seen, sometimes, it returns in two set, but not always.

Reply:
I am not sure I understand what you mean by the callback is called in multiple sets. The callback reports change in the query result, and the result will not be separated in multiple set.

Reply:
I will go through the code once again.

Reply:
You are right, my wrong understanding. Will revert and see, if the issue with unit test keep happening.

Reply:
reverted the change!